### DISCUSSION TITLE
error: Package subpath './public/extractFiles' is not defined by "exports" in /home/bot/wiki/node_modules/extract-files/package.json

### DISCUSSION THREAD

**Discussion by AzpektDev** [requarks/wiki#5113] [Error / Bug Report] [answered]
How i can fix that error:
```error: Package subpath './public/extractFiles' is not defined by "exports" in /home/bot/wiki/node_modules/extract-files/package.json```

**Comment by tigerblue77** [comment, marked as answer]
Can you give more details ? Which version of Wiki.js are you trying to install ? Which version of nodeJS are you using ?
I had this error using nodeJS 18 (latest stable version) which seems not supported by Wiki.js yet.
=> Downgrade to nodeJS 16

Sources :
- https://github.com/requarks/wiki/discussions/4869
- https://docs.requarks.io/install/requirements

**Reply by AzpektDev** [reply]
I was using NodeJS 18 too. Degrading to earlier versions helped.

**Reply by tigerblue77** [reply]
You're welcome.

**Reply by joshuaherr** [reply]
Hello,
I ran into the exact same problem launching a new installation of Wiki js. Following tigerblue77's solution (downgrading to version 16 of node js) fixed the problem for me as well. Thanks for asking the question AzpektDev and thanks for answering tigerblue77!

**Reply by zababurinsv** [reply]
I get this error with last wersion wiki.js. Can i fix it with node js 18.0 ?

**Reply by program-spiritual** [reply]
good, I meet the same question, you save me time, Thanx!

**Reply by lgarcia-sfisas** [reply]
You should update these repositories, we are already on version 18.14.0 of NodeJS.

**Comment by bruno-at-bareos** [comment]
Maybe it is time to upgrade extract-files (version 9.0 in wikijs) when last is 13.0 supporting newer nodejs version.

**Comment by tigerfoot** [comment]
For those who want to run nodejs16 under openSUSE Tumbleweed you can still install it from [nodejs repository](
https://download.opensuse.org/repositories/devel:/languages:/nodejs/openSUSE_Tumbleweed/x86_64/)
then you can trick the service by specifying the /usr/bin/nodejs16 executable.

**Comment by hinteadan** [comment]
This is still an issue today 2023-10-11, even though the official page states that both nodeJS 18 and 20 are supported with the latest wiki version! (https://docs.requarks.io/install/requirements#nodejs). Nasty, not cool... 😞

**Reply by k1llerk3ks** [reply]
I have barely any knowledge abot node or javascript in general, but i was able to fix it with the help of this file:

https://github.com/requarks/wiki/blob/fd91caff1da1683473fc3b65ecab84a41f4ebb8a/patches/extract-files%2B9.0.0.patch#L9

i was running the container, which failed at some point (always), adding those three lines brought him back up nicely again.

**Reply by tigerfoot** [reply]
I've today use 2.5.300 version, installed nodejs20 (instead my old 16) and a simple restart make wikijs just working.
So thanks I completely missed in August update the fact that 20 was supported.

**Reply by hinteadan** [reply]
Hmm, I only tried Node18. I'll give it a go with v20 if U say it works in your case... 🤞

**Reply by hinteadan** [reply]
> I have barely any knowledge abot node or javascript in general, but i was able to fix it with the help of this file:
> 
> https://github.com/requarks/wiki/blob/fd91caff1da1683473fc3b65ecab84a41f4ebb8a/patches/extract-files%2B9.0.0.patch#L9
> 
> i was running the container, which failed at some point (always), adding those three lines brought him back up nicely again. 

But those 3 lines are already part of the change set of the latest release... They're already there if you update to the latest version. Which is the case for me; and I still get the error _(with node18)..._ 🤷‍♂️

**Reply by darkjester-spacehamster** [reply]
I am also getting this error, trying to install:

wikijs 2.5.300
node.js 20.9.0

![image](https://github.com/requarks/wiki/assets/103436900/c2429a68-f60b-4607-abcc-b3232ab5a1ba)

**Reply by phreaker0** [reply]
Getting the same error with nodejs 18 and 20 with the latest packaged release version v2.5.301 from:
https://github.com/requarks/wiki/releases/download/v2.5.301/wiki-js.tar.gz

**Comment by zieglerjm** [comment]
Is this still not fixed? Using NodeJS v.20.11.1.

"error: Package subpath './public/extractFiles' is not defined by "exports" in E:\repos\WikiJS\node_modules\extract-files\package.json"

NodeJS v18.19.1 has the same error.
Looks like v16.20.2 is working for me.

**Comment by amarg3891** [comment]
@zieglerjm @wngdarkjester @hinteadan @AzpektDev  you guys dont have to do anything just navigate to your node_modules :- 
and track 3 files which is below - 
1. node_modules/@graphql-tools/links/node_modules/apollo-upload-client/public/createUploadLink.js
2. node_modules/@graphql-tools/links/node_modules/apollo-upload-client/public/isExtractableFile.js
3. node_modules/@graphql-tools/links/node_modules/apollo-upload-client/public/ReactNativeFile.js

and correct path of extract-files module like below -

var extractFiles = require('../../extract-files/public/extractFiles');
module.exports = require('../../extract-files/public/isExtractableFile');
module.exports = require('../../extract-files/public/ReactNativeFile');
 
 After doing that it works like a charm.
 
 Used Node Latest Version - 20.11.1
 
 
![image](https://github.com/requarks/wiki/assets/63994344/1b6e9486-740c-4c2d-9328-b5483932dee2)

**Reply by simone80an** [reply]
This worked for Node 22 too! Thanks.

**Reply by quasiblob** [reply]
I'm facing the same issue, but I don't even find such files in my node_modules so how to fix this. 

Has something changed recently related to this issue or have these files been replaced by something else?

**Reply by giri-qs** [reply]
> I'm facing the same issue, but I don't even find such files in my node_modules so how to fix this.
> 
> Has something changed recently related to this issue or have these files been replaced by something else?

node_modules/apollo-upload-client -> update on this path  with node 22

**Reply by sebastian-gajda** [reply]
@giri-qs, is it working now on node 22?

**Reply by sebastian-gajda** [reply]
> > I'm facing the same issue, but I don't even find such files in my node_modules so how to fix this.
> > Has something changed recently related to this issue or have these files been replaced by something else?
> 
> node_modules/apollo-upload-client -> update on this path with node 22

Is it working now?
I use node 22 but not working

**Reply by giri-qs** [reply]
> > > I'm facing the same issue, but I don't even find such files in my node_modules so how to fix this.
> > > Has something changed recently related to this issue or have these files been replaced by something else?
> > 
> > 
> > node_modules/apollo-upload-client -> update on this path with node 22
> 
> Is it working now? I use node 22 but not working

It is working . The search the files mentioned. You can fuzzy also and alter the path manually. That's it.

**Reply by vsiddharthv** [reply]
Working as mentioned by @amarg3891 ! 
I faced the same issue with Wiki.js v2.5.308 and node v22+
For more clarity, the files were here:
<img width="989" height="682" alt="screenshot_20251013_105149_279" src="https://github.com/user-attachments/assets/5a5d6a5f-10de-4176-a68c-49efae108f83" />

Update the path in all three of these files
